### PR TITLE
remove ParthenonInit and add docs

### LINKED ISSUE #929: Does anyone use `ParthenonInit`
For historical reasons, we have two ways of setting up Parthenon's runtime:
1. First call `ParthenonInitEnv` and then call` ParthenonInitPackagesAndMesh`. The former sets up MPI and Kokkos, and reads in the command line arguments and input deck. The latter calls all the `Initilalize` function for all packages and calls the mesh constructor.
2. `ParthenonInit` simply calls `ParthenonInitEnv` and `ParthenonInitPackagesAndMesh` one after the other.

It's easy to confuse `ParthenonInit` and `ParthenonInitPackagesAndMesh` by writing code like:
```C++
ParthenonManager pman;
pman.ParthenonInit(argc, argv);

// do some code, set app inputs

pman.ParthenonInitPackagesAndMesh();
```
where the intended code was actually

```C++
ParthenonManager pman;
pman.ParthenonInitEnv(argc, argv);

// do some code, set app inputs

pman.ParthenonInitPackagesAndMesh();
```

This seems error prone. Does anyone actually use the older, unified `ParthenonInit` function? If not, perhaps we should just remove it. 

Alternatively, we could add error checking so that if any of the `ParthenonInit*` functions is called twice by the same manager object, an error is raised.

### PR DESCRIPTION
**Pull request by Yurlungur**
## PR Summary

Based on the discussion in #929 , it seems like it might be a good idea to remove the ``ParthenonManager::ParthenonInit`` function. This PR does so. I also add documentation on ``ParthenonManager``, which I couldn't otherwise find.

 **This is a breaking change.** I would like the go-ahead from downstream codes before merging this.

## PR Checklist

- [x] Code passes cpplint
- [x] New features are documented.
- [ ] Adds a test for any bugs fixed. Adds tests for new features.
- [x] Code is formatted
- [x] Changes are summarized in CHANGELOG.md
- [ ] CI has been triggered on [Darwin](https://re-git.lanl.gov/eap-oss/parthenon/-/pipelines) for performance regression tests.
- [x] Docs build
- [x] (@lanl.gov employees) Update copyright on changed files

### CODE REVIEW

File: src/parthenon_manager.cpp [hunk 1]
```diff
@@ -40,16 +40,12 @@
 
 namespace parthenon {
 
-ParthenonStatus ParthenonManager::ParthenonInit(int argc, char *argv[]) {
-  auto manager_status = ParthenonInitEnv(argc, argv);
-  if (manager_status != ParthenonStatus::ok) {
-    return manager_status;
+ParthenonStatus ParthenonManager::ParthenonInitEnv(int argc, char *argv[]) {
+  if (called_init_env_) {
+    PARTHENON_THROW("ParthenonInitEnv called twice!");
```
Comment (lroberts36) [inline src/parthenon_manager.cpp:45]: nice.

File: src/parthenon_manager.cpp [hunk 2]
```diff
@@ -146,6 +142,11 @@ ParthenonStatus ParthenonManager::ParthenonInitEnv(int argc, char *argv[]) {
 }
 
 void ParthenonManager::ParthenonInitPackagesAndMesh() {
+  if (called_init_packages_and_mesh_) {
+    PARTHENON_THROW("Called ParthenonInitPackagesAndMesh twice!");
+  }
+  called_init_packages_and_mesh_ = true;
+
```
Comment (lroberts36) [inline src/parthenon_manager.cpp:149]: nice.

**Review by lroberts36 — APPROVED** [review]
Thanks, LGTM.

**Review by c-prather — APPROVED** [review]
LGTM.  I'd have done static vars with function scope but this is more flexible.  Could see e.g. erroring in driver.Execute if init wasn't called, or other hand-holding like that if there are common errors.  It also allows multiple `ParthenonManager`s I guess, though I'm not sure MPI & Kokkos would react well to that anyway.

For my own edification: are there other reasons to prefer members?

**Comment by Yurlungur** [comment]
> LGTM. I'd have done static vars with function scope but this is more flexible. Could see e.g. erroring in driver.Execute if init wasn't called, or other hand-holding like that if there are common errors. It also allows multiple `ParthenonManager`s I guess, though I'm not sure MPI & Kokkos would react well to that anyway.
> 
> For my own edification: are there other reasons to prefer members?

To be honest, I didn't think about it very hard. Members seemed natural and easy to reason about, since the object is already stateful. If people prefer statics, that's fine with me.

**Comment by Yurlungur** [comment]
Ping someone from AthenaPK @BenWibking @forrestglines @pgrete . Can we merge this in?

File: src/parthenon_manager.cpp [hunk 3]
```diff
@@ -40,16 +40,12 @@
...
   }
-  ParthenonInitPackagesAndMesh();
-  return ParthenonStatus::ok;
-}
+  called_init_env_ = true;
 
-ParthenonStatus ParthenonManager::ParthenonInitEnv(int argc, char *argv[]) {
   // initialize MPI
 #ifdef MPI_PARALLEL
   if (MPI_SUCCESS != MPI_Init(&argc, &argv)) {
```
Comment (forrestglines) [inline src/parthenon_manager.cpp:51]: At some point we'd like to optionally start with`MPI_Init_thread` which is apparently required by HDF5 subfiling. Another time and PR though

**Review by forrestglines — APPROVED** [review]
I can confirm that this PR works for AthenaPK (Had some unrelated trouble with ctest on Frontier since apparently the python script couldn't find the generated outputs in the filesystem).

**Pull request merged**